lib/taskana-core/src/main/java/pro/taskana/spi/priority/internal/PriorityServiceManager.java

```diff
@@ -0,0 +1,88 @@
+package pro.taskana.spi.priority.internal;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import pro.taskana.common.api.exceptions.SystemException;
+import pro.taskana.common.internal.util.LogSanitizer;
+import pro.taskana.spi.priority.api.PriorityServiceProvider;
+import pro.taskana.task.api.models.TaskSummary;
+
+public class PriorityServiceManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PriorityServiceManager.class);
+  private static PriorityServiceManager singleton;
+  private final ServiceLoader<PriorityServiceProvider> serviceLoader;
+  private boolean enabled = false;
+
+  private PriorityServiceManager() {
+    serviceLoader = ServiceLoader.load(PriorityServiceProvider.class);
+    for (PriorityServiceProvider priorityProvider : serviceLoader) {
+      LOGGER.info("Registered PriorityServiceProvider: {}", priorityProvider.getClass().getName());
+      enabled = true;
+    }
+    if (!enabled) {
+      LOGGER.info("No PriorityServiceProvider found. Running without PriorityServiceProvider.");
+    }
+  }
+
+  public static synchronized PriorityServiceManager getInstance() {
+    if (singleton == null) {
+      singleton = new PriorityServiceManager();
+    }
+    return singleton;
+  }
+
+  public static boolean isPriorityServiceEnabled() {
+    return Objects.nonNull(singleton) && singleton.enabled;
+  }
+
+  public Optional<Integer> calculatePriorityOfTask(TaskSummary task) {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Sending task to PriorityServiceProviders: {}", task);
+    }
+
```

Comment:
@benjamineckstein I changed back to a for-clause because it seemed as if I am only able to access the various providers inside the serviceLoader this way and not by a stream.